perun-core/src/main/java/cz/metacentrum/perun/core/impl/MembersManagerImpl.java

```diff
@@ -705,98 +707,32 @@ public MemberGroupStatus getUnifiedMemberGroupStatus(PerunSession sess, User use
+		String select = getSQLSelectForMembersPage(query);
+		String searchQuery = getSQLWhereForMembersPage(query);
 
-			namedParams = Utils.getMapSqlParameterSourceToSearchUsersOrMembers(query.getSearchString(), attributesToSearchBy);
-		}
 
 		namedParams.addValue("voId", vo.getId());
 		namedParams.addValue("offset", query.getOffset());
 		namedParams.addValue("limit", query.getPageSize());
 
-		String statusesQueryString = "";
-		if (query.getStatuses() != null && !query.getStatuses().isEmpty()) {
-			statusesQueryString = " AND members.status in (:statuses) ";
-			List<Integer> statusCodes = query.getStatuses().stream()
-				.map(Status::getCode)
-				.collect(Collectors.toList());
-			namedParams.addValue("statuses", statusCodes);
-		}
+		String statusesQueryString = getVoStatusSQLConditionForMembersPage(query, namedParams);
 
-		String groupQueryJoin = "";
-		String groupQueryWhere = "";
-		String groupStatusesQueryString = "";
-		if (query.getGroupId() != null) {
-			groupQueryJoin = " JOIN (SELECT group_id, member_id, min(source_group_status) as source_group_status, " +
-									" min(membership_type) as membership_type, null as source_group_id " +
-									" FROM groups_members " +
-									" WHERE group_id = (:groupId) " +
-									" GROUP BY group_id, member_id) as groups_members " +
-								" ON members.id=groups_members.member_id ";
-			groupQueryWhere = " AND groups_members.group_id = (:groupId) ";
-			namedParams.addValue("groupId", query.getGroupId());
+		String groupStatusesQueryString = getGroupStatusSQLConditionForMembersPage(query, namedParams);
 
-			if (query.getGroupStatuses() != null && !query.getGroupStatuses().isEmpty()) {
-				groupStatusesQueryString = " AND groups_members.source_group_status in (:groupStatuses) ";
-				List<Integer> groupStatusCodes = query.getGroupStatuses().stream()
-					.map(MemberGroupStatus::getCode)
-					.collect(Collectors.toList());
-				namedParams.addValue("groupStatuses", groupStatusCodes);
-			}
-		}
-
-		Paginated<Member> members = namedParameterJdbcTemplate.query(
-				"SELECT DISTINCT " +
-					(query.getGroupId() == null ? memberMappingSelectQuery : groupsMembersMappingSelectQuery) +
-				query.getSortColumn().getSqlSelect() +
-				", DENSE_RANK() OVER (order by members.id) + DENSE_RANK() OVER (ORDER BY members.id DESC) - 1 AS total_count" +
-				" FROM members " +
-				" LEFT JOIN users ON members.user_id=users.id " +
-				" LEFT JOIN user_ext_sources ues ON ues.user_id=users.id " +
-				query.getSortColumn().getSqlJoin() +
-				groupQueryJoin +
-				searchStringQueryJoin +
-				" WHERE members.vo_id = (:voId) " +
+		return namedParameterJdbcTemplate.query(
+			    select +
+				" WHERE members.vo_id = (:voId)" +
 				statusesQueryString +
-				groupQueryWhere +
 				groupStatusesQueryString +
-				searchStringQueryWhere +
-				" ORDER BY " + query.getSortColumn().getSqlOrderBy(query) +
+					searchQuery +
+			    " ORDER BY " + query.getSortColumn().getSqlOrderBy(query) +
```

Comment:
Wrong indentation

Reply:
Fixed.

Reply:
It is still the same :)

Reply:
Oh, I see. You meant spaces instead of a tab. That should be fixed now.